sdcm/nemesis.py

```diff
@@ -1115,6 +1115,41 @@ def disrupt_network_start_stop_interface(self):  # pylint: disable=invalid-name
         finally:
             self.target_node.remoter.run("sudo /sbin/ifup eth1")
 
+    def disrupt_streaming_error(self):
+        """
+        Stop decommission in middle to trigger some streaming fails, then rebuild the data on the node.
+        If the node is decommission unexpectedly, need to re-add a new node to cluster.
+        """
+        self._set_current_disruption('StreamingError')
+        target_node_ip = self.target_node.ip_address
+
+        def _decommission():
+            try:
+                self.target_node.run_nodetool("decommission")
+            except Exception:  # pylint: disable=broad-except
+                self.log.debug('Decommission is stopped')
+        log_start = self.target_node.mark_log()
+        decommission_thread = threading.Thread(target=_decommission)
+        decommission_thread.start()
+        time.sleep(10)
+        decommission_thread.join(0)
```

Comment:
1. need to join after  the `pkill` otherwise it is blocking and you will be stuck here until `nodetool decommission` ends
2. need to set reasonable timeout like 60 sec for join, and handle the timeout (probably `pkill -9` will be good way to stop)

Reply:
Thanks, will fix it.